app/controllers/catalog_controller.rb

```diff
@@ -41,6 +42,48 @@ def update_user_per_cat_page
     end
   end
 
+  def submit_form # rubocop:disable MethodLength, AbcSize
```

Comment:
Hmm.. I agree that it's not good that it duplicates code from update_cart, which hurts it in sustainability, but I feel like we should handle the dates in the form separately from the dates in the cart. Right now submitting the form with an error in dates doesn't affect the original cart, so the original values of the cart are restored. Integrating update_cart into it to handle the date validation on the form removes that function because it sets the dates back to their default values. I'm not sure which is more important in this case. 
Another option is to call update_cart again after the validations on the form to handle the blackouts, but I'm not sure if it's worth it to call it just to handle the blackout notifications.